prow/plugins/trigger/ic.go

```diff
@@ -134,13 +159,15 @@ func handleIC(c client, trustedOrg string, ic github.IssueCommentEvent) error {
 
 	var errors []error
 	for _, job := range requestedJobs {
-		if !job.RunsAgainstBranch(pr.Base.Ref) {
-			if err := c.GitHubClient.CreateStatus(org, repo, pr.Head.SHA, github.Status{
-				State:       github.StatusSuccess,
-				Context:     job.Context,
-				Description: "Skipped",
-			}); err != nil {
-				return err
+		if !job.RunsAgainstBranch(pr.Base.Ref) || (!job.AlwaysRun && job.RunIfChanged != "" && !job.RunsAgainstChanges(changedFiles)) {
+			if !job.SkipReport {
```

Comment:
Can't you inline this into the parent if clause?

Reply:
There is a `continue` statement that should be executed regardless of the result of `!job.SkipReport`.